### ISSUE TITLE
class_error: 100

### ISSUE THREAD

**Issue by doilion** [PKU-ICST-MIPL/SIA-OVD_ACMMM2024#1]
![Image](https://github.com/user-attachments/assets/2abebbf4-e20a-4820-9d8d-d0cb4f7a32d2)

Hello, I have met a problem. The class_error in the test phase is always 100, and I can do nothing to change this.

**Comment by wangzishuo029** [comment]
May I ask that did you encounter this problem during model training, or during testing after convergence? If this problem happens during testing, is it tested on the ground-truth bounding-boxes or predicted bounding-boxes?

**Comment by doilion** [comment]
I encountered this issue during the training process. After training for about 5-6 epochs, I noticed that the results remained identical. I tested it using a small COCO dataset called "tiny_coco" with approximately 50 images and observed the same problem. The batch size was set to 2, epochs to 35, and the learning rate was set to 1e-4. I was training on my own dataset and didn’t encounter any issues when training the "region" part. I’m unsure why this is happening—could it be because I didn’t load the pre-trained weights?

**Comment by doilion** [comment]
I encountered this issue during the training process. After training for about 5-6 epochs, I noticed that the results remained identical. I tested it using a small COCO dataset called "tiny_coco" with approximately 50 images and observed the same problem. The batch size was set to 2, epochs to 35, and the learning rate was set to 1e-4. I was training on my own dataset and didn’t encounter any issues when training the "region" part. I’m unsure why this is happening—could it be because I didn’t load the pre-trained weights?

**Comment by wangzishuo029** [comment]
I think there might be a problem with the code's support for the dataset. Again, is this a problem both on your own dataset and tiny_coco? Is there such a problem with the original COCO dataset?

**Comment by wangzishuo029** [comment]
Please refer to [CocoEvaluator](https://github.com/PKU-ICST-MIPL/SIA-OVD_ACMMM2024/blob/458c62312257451e1938136b2d7a06024e1e1374/datasets/coco_eval.py#L26) in ``dataset/coco_eval.py``, and make sure that you have correctly set ``base_catids`` and ``target_catids`` according to your own dataset.

**Comment by doilion** [comment]
I haven't tried the original COCO dataset yet, but I'll test it when I have time. I've already modified the `coco.py`  and `coco_eval.py`  file and am not using the OVD approach. I'm curious, do you all train your object detectors from scratch, or do you fine-tune them?

**Comment by wangzishuo029** [comment]
The object detectors (DAB-DETR) is trained from scratch in the second training phase (after the first training phase when the adapters are optimized).

Do you mean that your base classes and target classes are the same? If so, I recommend that you use a traditional object detectors rather than open-vocabulary object detectors.
